lib/utils/process-jobs.ts

```diff
@@ -46,9 +46,10 @@ export const processJobs = function(this: Agenda, extraJob: Job) {
    * @param name name of job to check if we should lock or not
    * @returns whether or not you should lock job
    */
-  function shouldLock(name: string): boolean {
+  function shouldLock(name: string, jobId: string = ''): boolean {
```

Comment:
The JSDoc is lacking description of the `jobId`.

Having the `jobId` implies that `shouldLock()` assesses this exact job ID. But it's not. The `jobId` is used for logging only.
I'd recommend moving the `debug()` to the place where `shouldLock()` is being called, and leaving this method intact.

Reply:
removed the jobId